lib/heroku-auth-token.js

```diff
@@ -1,21 +1,35 @@
 'use strict';
 
 let shellpromise = require('shellpromise');
+let authToken;
 
-module.exports = function () {
+const getAuthFromCli = () => {
+	return shellpromise('heroku auth:whoami 2>/dev/null')
+		.then(function () {
+			return shellpromise('heroku auth:token 2>/dev/null');
+		})
+		.then(function (token) {
+			return token.trim();
+		})
+		.catch(function (err) {
+			console.error(err); // eslint-disable-line no-console
+			throw new Error('Please make sure the Heroku CLI is authenticated by running `heroku auth:token`');
+		});
+};
+
+module.exports = async function () {
 	if (process.env.HEROKU_AUTH_TOKEN) {
+
 		return Promise.resolve(process.env.HEROKU_AUTH_TOKEN);
+
 	} else {
-		return shellpromise('heroku auth:whoami 2>/dev/null')
-			.then(function () {
-				return shellpromise('heroku auth:token 2>/dev/null');
-			})
-			.then(function (token) {
-				return token.trim();
-			})
-			.catch(function (err) {
-				console.error(err); // eslint-disable-line no-console
-				throw new Error('Please make sure the Heroku CLI is authenticated by running `heroku auth:token`');
-			});
+
+		if (authToken) {
+			return authToken;
```

Comment:
If you are going to store it, you could just use `process.env.HEROKU_AUTH_TOKEN`, and save on having another nested `if` statement ie,

    process.env.HEROKU_AUTH_TOKEN = await getAuthFromCli();

Setting `process.env.HEROKU_AUTH_TOKEN` inside this script wouldn't persist outside of this script being run too.